PalasoUIWindowsForms/Reporting/MemoryManagement.cs

```diff
@@ -52,30 +54,49 @@ public static bool CheckMemory(bool minor, string eventDescription, bool okToBot
 				var meminfo = File.ReadAllText("/proc/meminfo");
 				var match = new Regex(@"MemTotal:\s+(\d+) kB").Match(meminfo);
 				if (match.Success)
-				{
 					totalPhysicalMemory = ulong.Parse(match.Groups[1].Value) * 1024;
+				ulong totalSwapMemory = 0;
+				var match2 = new Regex(@"SwapTotal:\s+(\d+) kB").Match(meminfo);
+				if (match2.Success)
+					totalSwapMemory = ulong.Parse(match2.Groups [1].Value) * 1024;
+				var availableMemory = totalPhysicalMemory + totalSwapMemory;
+				if (is64BitProcess)
+				{
+					totalVirtualMemory = (availableMemory / 1024).ToString ("N0");
+				}
+				else
+				{
+					ulong twoGB = 2147483648L;
+					totalVirtualMemory = ((availableMemory > twoGB ? twoGB : availableMemory) / 1024).ToString("N0");
 				}
-				// So far we have no way to get anything corresponding to Windows's idea of the total virtual memory
-				// the process can use.
 			}
 			using (var proc = Process.GetCurrentProcess())
 			{
 				memorySize64 = proc.PrivateMemorySize64;
+				workingSet64 = proc.WorkingSet64;
 				message =
 					string.Format(
-						"{0}: total memory {1:N0}K, working set {2:N0}K, heap {3:N0}K, system physical {4:N0}K, system virtual {5}K",
+						"{0}: total memory {1:N0}K, working set {2:N0}K, heap {3:N0}K, system physical {4:N0}K, system virtual {5}K; {6}-bit process",
 						eventDescription,
 						memorySize64/1024,
-						proc.WorkingSet64/1024,
+						workingSet64/1024,
 						heapMem/1024,
 						totalPhysicalMemory/1024,
-						totalVirtualMemory);
+						totalVirtualMemory,
+						is64BitProcess ? 64 : 32);
 			}
 			if (minor)
 				Logger.WriteMinorEvent(message);
 			else
 				Logger.WriteEvent(message);
-			var danger = memorySize64 > 1000000000;
+			long safelimit = 1000000000;
+			if (is64BitProcess && totalPhysicalMemory >= 8192000000L)
+				safelimit = 2000000000;
+			bool danger = false;
+			if (Platform.IsWindows)
```

Comment:
This seems to deserve a comment explaining that we've found memorySize64 on Linux to be useless as a measure of RAM in use by the process.